projects/plugins/jetpack/tests/e2e/lib/env/playwright-environment.js

```diff
@@ -0,0 +1,155 @@
+const PlaywrightEnvironment = require( 'jest-playwright-preset/lib/PlaywrightEnvironment' ).default;
+const fs = require( 'fs' );
+const logger = require( '../logger' ).default;
+
+class PlaywrightCustomEnvironment extends PlaywrightEnvironment {
+	async setup() {
+		await super.setup();
+	}
+
+	async teardown() {
+		await super.teardown();
+	}
+
+	async handleTestEvent( event ) {
+		await super.handleTestEvent( event );
+
+		let testName = 'unknown_test';
+		let hookName = 'unknown_hook';
+
+		if ( event.test ) {
+			testName = `${ event.test.parent.name } - ${ event.test.name }`;
+		}
+
+		if ( event.hook ) {
+			hookName = `${ event.hook.type } - ${ event.hook.parent.name }`;
+		}
+
+		switch ( event.name ) {
+			case 'setup':
+				break;
+			case 'add_hook':
+				break;
+			case 'add_test':
+				break;
+			case 'run_start':
+				break;
+			case 'test_skip':
+				break;
+			case 'test_todo':
+				break;
+			case 'start_describe_definition':
+				break;
+			case 'finish_describe_definition':
+				break;
+			case 'run_describe_start':
+				break;
+			case 'test_start':
+				logger.info( `START TEST: ${ testName }` );
+				// await this.storeVideoFileName( testName );
+				break;
+			case 'hook_start':
+				logger.info( `START: ${ hookName }` );
+				// await this.storeVideoFileName( hookName );
+				break;
+			case 'hook_success':
+				logger.info( `SUCCESS: ${ hookName }` );
+				break;
+			case 'hook_failure':
+				logger.info( `FAILED: ${ hookName }` );
+				logger.error( JSON.stringify( event.hook.errors ) );
+				logger.error( JSON.stringify( event.error ) );
+				await this.saveScreenshot( hookName );
+				await this.storeVideoFileName( hookName );
+				await this.logHTML( hookName );
+				await this.logFailureToSlack( event.hook.parent.name, event.hook.type, event.error );
+				break;
+			case 'test_fn_start':
+				break;
+			case 'test_fn_success':
+				logger.info( `TEST PASSED: ${ testName }` );
+				break;
+			case 'test_fn_failure':
+				logger.info( `FAILED TEST: ${ testName }` );
+				logger.error( JSON.stringify( event.test.errors ) );
+				logger.error( JSON.stringify( event.error ) );
+				await this.saveScreenshot( testName );
```

Comment:
Do we really need to save screenshots and videos when running tests locally? is it useful at all?

Reply:
I find it useful. I'm running them locally quite often without debug options and the screenshots are the first thing I'm checking in case of failure after the error messages. I see no harm in having them enabled by default. In time, if needed, we can add a switch.